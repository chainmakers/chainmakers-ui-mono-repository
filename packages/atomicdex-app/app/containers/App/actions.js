// @flow
/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 * https://github.com/redux-utilities/flux-standard-action
 */

import {
  LOGIN,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
  LOGOUT,
  LOAD_WITHDRAW,
  LOAD_WITHDRAW_SUCCESS,
  LOAD_WITHDRAW_ERROR,
  LOAD_SWAP_SUCCESS,
  ELECTRUM_LOAD,
  ELECTRUM_ADD,
  ELECTRUM_ADD_SUCCESS,
  ELECTRUM_ADD_ERROR,
  BALANCE_LOAD_ALL,
  BALANCE_LOAD,
  BALANCE_LOAD_SUCCESS,
  BALANCE_LOAD_ERROR
} from './constants';
import type {
  LoginError,
  AddElectrumPayload,
  AddElectrumSuccessPayload,
  LoadbalacePayload,
  LoadBalanceSuccessPayload
} from './schema';
import type { ErrorType } from '../schema';

export function login(passphrase: string) {
  return {
    type: LOGIN,
    payload: {
      passphrase
    }
  };
}

export function loginSuccess(user) {
  return {
    type: LOGIN_SUCCESS,
    payload: {
      user
    }
  };
}

export function loginError(error: LoginError) {
  return {
    type: LOGIN_ERROR,
    error
  };
}

export function logout() {
  return {
    type: LOGOUT
  };
}

export function loadWithdraw(payload: {
  amount: number,
  address: string,
  coin: string
}) {
  return {
    type: LOAD_WITHDRAW,
    payload
  };
}

export function loadWithdrawSuccess(payload: {
  amount: number,
  address: string,
  coin: string
}) {
  return {
    type: LOAD_WITHDRAW_SUCCESS,
    payload
  };
}

export function loadWithdrawError(
  payload: {
    amount: number,
    address: string,
    coin: string
  },
  message: string
) {
  return {
    type: LOAD_WITHDRAW_ERROR,
    payload,
    error: {
      message
    }
  };
}

// eslint-disable-next-line flowtype/no-weak-types
export function loadSwapSuccess(payload: Object) {
  return {
    type: LOAD_SWAP_SUCCESS,
    payload
  };
}

// -- //

export function loadElectrums() {
  return {
    type: ELECTRUM_LOAD
  };
}

export function addElectrum(payload: AddElectrumPayload) {
  return {
    type: ELECTRUM_ADD,
    payload
  };
}

export function addElectrumSuccess(payload: AddElectrumSuccessPayload) {
  return {
    type: ELECTRUM_ADD_SUCCESS,
    payload
  };
}

export function addElectrumError(error: ErrorType) {
  return {
    type: ELECTRUM_ADD_ERROR,
    error
  };
}

export function loadAllBalance() {
  return {
    type: BALANCE_LOAD_ALL
  };
}

export function loadBalance(payload: LoadbalacePayload) {
  return {
    type: BALANCE_LOAD,
    payload
  };
}

export function loadBalanceSuccess(payload: LoadBalanceSuccessPayload) {
  return {
    type: BALANCE_LOAD_SUCCESS,
    payload
  };
}

export function loadBalanceError(error: ErrorType) {
  return {
    type: BALANCE_LOAD_ERROR,
    error
  };
}
