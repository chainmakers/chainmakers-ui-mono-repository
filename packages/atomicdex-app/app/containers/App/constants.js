/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */
export const APP_STATE_NAME = 'global';

export const LOGIN = 'atomicapp/App/LOGIN';
export const LOGIN_SUCCESS = 'atomicapp/App/LOGIN_SUCCESS';
export const LOGIN_ERROR = 'atomicapp/App/LOGIN_ERROR';

export const LOGOUT = 'atomicapp/App/LOGOUT';

export const LOAD_WITHDRAW = 'atomicapp/App/LOAD_WITHDRAW';
export const LOAD_WITHDRAW_SUCCESS = 'atomicapp/App/LOAD_WITHDRAW_SUCCESS';
export const LOAD_WITHDRAW_ERROR = 'atomicapp/App/LOAD_WITHDRAW_ERROR';

export const LOAD_SWAP_SUCCESS = 'atomicapp/App/LOAD_SWAP_SUCCESS';

// electrum actions
export const ELECTRUM_LOAD = 'atomicapp/App/ELECTRUM_LOAD';
export const ELECTRUM_ADD = 'atomicapp/App/ELECTRUM_ADD';
export const ELECTRUM_ADD_SUCCESS = 'atomicapp/App/ELECTRUM_ADD_SUCCESS';
export const ELECTRUM_ADD_ERROR = 'atomicapp/App/ELECTRUM_ADD_ERROR';

// balance
export const BALANCE_LOAD_ALL = 'atomicapp/App/BALANCE_LOAD_ALL';
export const BALANCE_LOAD = 'atomicapp/App/BALANCE_LOAD';
export const BALANCE_LOAD_SUCCESS = 'atomicapp/App/BALANCE_LOAD_SUCCESS';
export const BALANCE_LOAD_ERROR = 'atomicapp/App/BALANCE_LOAD_ERROR';
