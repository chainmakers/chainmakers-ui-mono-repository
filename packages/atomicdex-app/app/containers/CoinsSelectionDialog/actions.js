// @flow
import {
  SELECT_COIN_MODAL_SEARCH,
  SELECT_COIN_MODAL_SEARCH_SUCCESS,
  SELECT_COIN_MODAL_SETUP_SEARCH_API,
  SELECT_COIN_MODAL_SETUP_SEARCH_API_SUCCESS,
  SEARCH_STATE_SKIP_CREATION
} from './constants';

export function searchSelectCoinModal(input: string) {
  return {
    type: SELECT_COIN_MODAL_SEARCH,
    payload: {
      input
    }
  };
}

export function searchSelectCoinModalSuccess(payload) {
  return {
    type: SELECT_COIN_MODAL_SEARCH_SUCCESS,
    payload
  };
}

export function setupSearchApiForSelectCoinModal() {
  return {
    type: SELECT_COIN_MODAL_SETUP_SEARCH_API
  };
}

export function setupSearchApiForSelectCoinModalSuccess(payload) {
  return {
    type: SELECT_COIN_MODAL_SETUP_SEARCH_API_SUCCESS,
    payload
  };
}

export function skipSearchStateCreation() {
  return {
    type: SEARCH_STATE_SKIP_CREATION
  };
}
