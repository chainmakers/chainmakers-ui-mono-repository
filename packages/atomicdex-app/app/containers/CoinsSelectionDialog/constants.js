// eslint-disable-next-line import/prefer-default-export
export const APP_STATE_NAME = 'coinsSelectionDialog';

// SEARCH STATUS
export const SEARCH_STATE_NULL =
  'atomicapp/CoinsSelectionDialog/SEARCH_STATE_NULL';
export const SEARCH_STATE_CREATE =
  'atomicapp/CoinsSelectionDialog/SEARCH_STATE_CREATE';
export const SEARCH_STATE_SKIP_CREATION =
  'atomicapp/CoinsSelectionDialog/SEARCH_STATE_SKIP_CREATION';
export const SEARCH_STATE_READY =
  'atomicapp/CoinsSelectionDialog/SEARCH_STATE_READY';

export const SELECT_COIN_MODAL_SEARCH =
  'atomicapp/CoinsSelectionDialog/SELECT_COIN_MODAL_SEARCH';
export const SELECT_COIN_MODAL_SEARCH_SUCCESS =
  'atomicapp/CoinsSelectionDialog/SELECT_COIN_MODAL_SEARCH_SUCCESS';
export const SELECT_COIN_MODAL_SETUP_SEARCH_API =
  'atomicapp/CoinsSelectionDialog/SELECT_COIN_MODAL_SETUP_SEARCH_API';
export const SELECT_COIN_MODAL_SETUP_SEARCH_API_SUCCESS =
  'atomicapp/CoinsSelectionDialog/SELECT_COIN_MODAL_SETUP_SEARCH_API_SUCCESS';
export const SELECT_COIN_MODAL_SETUP_SEARCH_API_ERROR =
  'atomicapp/CoinsSelectionDialog/SELECT_COIN_MODAL_SETUP_SEARCH_API_ERROR';
