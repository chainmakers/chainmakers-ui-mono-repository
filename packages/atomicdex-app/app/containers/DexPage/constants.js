// eslint-disable-next-line import/prefer-default-export
export const APP_STATE_NAME = 'dex';
export const LOAD_PRICES = 'atomicapp/DexPage/LOAD_PRICES';
export const LOAD_PRICES_SUCCESS = 'atomicapp/DexPage/LOAD_PRICES_SUCCESS';
export const LOAD_PRICES_ERROR = 'atomicapp/DexPage/LOAD_PRICES_ERROR';
export const LOAD_PRICE = 'atomicapp/DexPage/LOAD_PRICE';
export const LOAD_BEST_PRICE = 'atomicapp/DexPage/LOAD_BEST_PRICE';
export const LOAD_BUY_COIN = 'atomicapp/DexPage/LOAD_BUY_COIN';
export const LOAD_BUY_COIN_SUCCESS = 'atomicapp/DexPage/LOAD_BUY_COIN_SUCCESS';
export const LOAD_BUY_COIN_ERROR = 'atomicapp/DexPage/LOAD_BUY_COIN_ERROR';
export const CLEAR_BUY_COIN_ERROR = 'atomicapp/DexPage/CLEAR_BUY_COIN_ERROR';
export const LOAD_RECENT_SWAPS = 'atomicapp/DexPage/LOAD_RECENT_SWAPS';
export const LOAD_RECENT_SWAPS_COIN =
  'atomicapp/DexPage/LOAD_RECENT_SWAPS_COIN';
export const LOAD_RECENT_SWAPS_SUCCESS =
  'atomicapp/DexPage/LOAD_RECENT_SWAPS_SUCCESS';
export const LOAD_RECENT_SWAPS_DATA_FROM_WEBSOCKET =
  'atomicapp/DexPage/LOAD_SWAP_DATA_FROM_WEBSOCKET';
export const LOAD_RECENT_SWAPS_ERROR =
  'atomicapp/DexPage/LOAD_RECENT_SWAPS_ERROR';
export const CHECK_TIMEOUT_EVENT = 'atomicapp/DexPage/CHECK_TIMEOUT_EVENT';
export const CHECK_UPDATE_SWAP_EVENT =
  'atomicapp/DexPage/CHECK_UPDATE_SWAP_EVENT';
export const AUTO_HIDE_SNACKBAR_TIME = 6000;
export const TIME_LOOP = 5 * 1000;

// https://github.com/KomodoPlatform/Documentation/blob/master/docs/source/barterDEX/MarketmakerErrors.rst#marketmaker-error-codes
export const SWAP_TX_DEFAULT =
  '0000000000000000000000000000000000000000000000000000000000000000';

export const SWAP_TIMEOUT = 'atomicapp/DexPage/SWAP_TIMEOUT';
export const SWAP_MAKE_A_NEW = 'atomicapp/DexPage/SWAP_MAKE_A_NEW';
export const SWAP_DETAIL_MODAL_OPEN =
  'atomicapp/DexPage/SWAP_DETAIL_MODAL_OPEN';
export const SWAP_DETAIL_MODAL_CLOSE =
  'atomicapp/DexPage/SWAP_DETAIL_MODAL_CLOSE';

export const SELECT_COIN_MODAL_OPEN =
  'atomicapp/DexPage/SELECT_COIN_MODAL_OPEN';
export const SELECT_COIN_MODAL_CLOSE =
  'atomicapp/DexPage/SELECT_COIN_MODAL_CLOSE';
export const SELECT_COIN_MODAL_CLICK =
  'atomicapp/DexPage/SELECT_COIN_MODAL_CLICK';

export const COIN_PAYMENT_SELECT = 'atomicapp/DexPage/COIN_PAYMENT_SELECT';

export const NA = 'N/A';

export const DEXFEE = 777;
export const NUMCOIN = 100000000;
export const BUFF_PRICES = 1.01 * NUMCOIN; // 1%
// export const BUFF_PRICES = 1.0 * NUMCOIN; // 0%

// SEARCH STATUS
export const SEARCH_STATE_NULL = 'atomicapp/DexPage/SEARCH_STATE_NULL';
export const SEARCH_STATE_CREATE = 'atomicapp/DexPage/SEARCH_STATE_CREATE';
export const SEARCH_STATE_READY = 'atomicapp/DexPage/SEARCH_STATE_READY';

// Joyride actions
export const JOYRIDE_OPEN = 'atomicapp/DexPage/JOYRIDE_OPEN';
export const JOYRIDE_CLOSE = 'atomicapp/DexPage/JOYRIDE_CLOSE';
