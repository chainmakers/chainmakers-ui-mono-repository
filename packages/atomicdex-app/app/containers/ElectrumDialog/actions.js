// @flow
import {
  ELECTRUM_DIALOG_SHOW,
  ELECTRUM_DIALOG_HIDE,
  ELECTRUM_ADD
} from './constants';
import type { AddElectrumPayload } from '';

export function showElectrumDialog() {
  return {
    type: ELECTRUM_DIALOG_SHOW
  };
}

export function hideElectrumDialog() {
  return {
    type: ELECTRUM_DIALOG_HIDE
  };
}

export function addElectrum(payload: AddElectrumPayload) {
  return {
    type: ELECTRUM_ADD,
    payload
  };
}
