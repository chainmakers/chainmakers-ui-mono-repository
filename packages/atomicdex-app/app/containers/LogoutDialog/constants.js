// eslint-disable-next-line import/prefer-default-export
export const APP_STATE_NAME = 'logoutDialog';
export const LOGOUT_DIALOG_SHOW = 'atomicapp/LogoutDialog/LOGOUT_DIALOG_SHOW';
export const LOGOUT_DIALOG_HIDE = 'atomicapp/LogoutDialog/LOGOUT_DIALOG_HIDE';
