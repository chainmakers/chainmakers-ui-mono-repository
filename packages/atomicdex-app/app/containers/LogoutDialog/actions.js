import { LOGOUT_DIALOG_SHOW, LOGOUT_DIALOG_HIDE } from './constants';

export function showLogoutDialog() {
  return {
    type: LOGOUT_DIALOG_SHOW
  };
}

export function hideLogoutDialog() {
  return {
    type: LOGOUT_DIALOG_HIDE
  };
}
