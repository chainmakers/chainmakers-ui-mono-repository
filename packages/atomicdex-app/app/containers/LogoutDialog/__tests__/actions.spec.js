import { showLogoutDialog, hideLogoutDialog } from '../actions';
import { LOGOUT_DIALOG_SHOW, LOGOUT_DIALOG_HIDE } from '../constants';

describe('containers/LogoutDialog/actions/showLogoutDialog', () => {
  it('should showLogoutDialog should create showLogoutDialog action', () => {
    expect(showLogoutDialog()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOGOUT_DIALOG_SHOW
    };

    expect(showLogoutDialog()).toEqual(expectedResult);
  });
});

describe('containers/LogoutDialog/actions/hideLogoutDialog', () => {
  it('should hideLogoutDialog should create hideLogoutDialog action', () => {
    expect(hideLogoutDialog()).toMatchSnapshot();
  });

  it('should return the correct type and the passed name', () => {
    const expectedResult = {
      type: LOGOUT_DIALOG_HIDE
    };

    expect(hideLogoutDialog()).toEqual(expectedResult);
  });
});
