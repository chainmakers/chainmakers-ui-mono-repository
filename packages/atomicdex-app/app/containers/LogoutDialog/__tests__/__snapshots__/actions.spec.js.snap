// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/LogoutDialog/actions/hideLogoutDialog should hideLogoutDialog should create hideLogoutDialog action 1`] = `
Object {
  "type": "atomicapp/LogoutDialog/LOGOUT_DIALOG_HIDE",
}
`;

exports[`containers/LogoutDialog/actions/showLogoutDialog should showLogoutDialog should create showLogoutDialog action 1`] = `
Object {
  "type": "atomicapp/LogoutDialog/LOGOUT_DIALOG_SHOW",
}
`;
